skyportal/plot.py

```diff
@@ -799,6 +946,26 @@ def make_legend_items_and_detection_lines(
         instrument=[],
         stacked=[],
     )
+    unique_filters = list(data['filter'].unique())
+    unique_origins = list(data['origin'].unique())
+    # I don't care about None.
+    if 'None' in unique_origins:
+        unique_origins.remove('None')
+
+    show_all_filters = True
+    show_all_origins = True
+
+    if user.preferences:
+        if "automaticallyVisibleFilters" in user.preferences:
+            for filter in user.preferences['automaticallyVisibleFilters']:
+                if filter in unique_filters:
+                    show_all_filters = False
+                    break
+        if "automaticallyVisibleOrigins" in user.preferences:
+            for origin in user.preferences['automaticallyVisibleOrigins']:
+                if origin in unique_origins:
+                    show_all_origins = False
+                    break
```

Comment:
I am wondering if it is not better to put this logic into the `check_visibility_on_phot_plot` function? That way we will have all the preferences fetching and hide/display choices done in one place.

Reply:
Is it just that this test is shared by multiple calls to functions?

Reply:
Another option would be to get a list of all the filters and origins to display and pass a list of indices. I don't know if that is better.

Reply:
yes, the test is shared by multiple calls to make_scatter, so I think this is an appropriate spot for it (and avoid doing the check over and over again in `check_visibility_on_phot_plot`). I have extracted the logic into a helper function though, it does stick out a bit in the middle of that function. The purpose of this code is the case where the plot does not contain any of the user's preferred filters/origins, in that case every point should be shown (I guess instead of none being shown, which is what would happen if there were no `show_all_filters` and `show_all_origins` flag and we just toggled visiblity based on if the point is in the user's preferences or not). so I would be inclined to keep the logic like this. are you okay with that @guynir42 ?

Reply:
Yeah this is fine.

Reply:
I think failures are unrelated, ok to merge? @guynir42

Reply:
Go ahead.